diff --git a/duckdb.cpp b/duckdb.cpp
--- a/duckdb.cpp
+++ b/duckdb.cpp
@@ -53722,8 +53722,12 @@
-
+
 #include <windows.h>
-
+
 #undef CreateDirectory
+#ifdef interface
+#undef interface
+#endif
+
 #undef MoveFile
 #undef RemoveDirectory
-
+
 #endif
@@ -337545,8 +337549,11 @@
 #endif
-
+
 #if defined(_WIN32)
 #include <windows.h>
+#ifdef interface
+#undef interface
+#endif
 #elif defined(__GNUC__)
 #include <sched.h>
 #include <unistd.h>
 #if defined(__GLIBC__)
@@ -466168,8 +466175,11 @@
 #else
 #ifdef _WIN32
 // Requires Windows Vista or Windows Server 2008 at minimum.
 #include <windows.h>
+#ifdef interface
+#undef interface
+#endif
 #if defined(WINVER) && WINVER >= 0x0600
 #define MUTEX_IS_WIN32_SRWLOCK
 #endif
 #else
@@ -595996,8 +596006,11 @@
 #include <string.h>
-
+
 #if defined(_WIN32)
 #include <windows.h>
+#ifdef interface
+#undef interface
+#endif
 #endif
-
+
 // Detect platforms known to support explicit_bzero()
 #if defined(__GLIBC__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 25)
@@ -596208,8 +596221,11 @@
 }
 #elif defined(_WIN32) || defined(WIN32) || defined(__CYGWIN__) || \
     defined(__MINGW32__) || defined(_WIN64)
 #include <windows.h>
+#ifdef interface
+#undef interface
+#endif
 mbedtls_ms_time_t mbedtls_ms_time(void)
 {
     FILETIME ct;
     mbedtls_ms_time_t current_ms;
@@ -615539,8 +615555,11 @@
 #endif
-
+
 #undef ERROR   /* reported already defined on VS 2015 (Rich Geldreich) */
 #include <windows.h>
+#ifdef interface
+#undef interface
+#endif
 #undef ERROR
 #define ERROR(name) ZSTD_ERROR(name)
-
+
 /* mutex */
